### DISCUSSION TITLE
Habitats and Biotopes - Code list "referenceHabitatTypeId

### DISCUSSION THREAD

**Discussion by Dnoack49** [INSPIRE-MIF/helpdesk-validator#796] [Q&A] [answered]
Dear all,

I validated the following HB-Dataset:
https://inspire.brandenburg.de/services/hb_bbk_wfs?service=WFS&version=2.0.0&request=GetFeature&typenames=hb:Habitat&Count=10 

For the codelist **referenceHabitatTypeId**  the _production Instance_ throws an error: 

![grafik](https://user-images.githubusercontent.com/8046960/175025906-5a46ec95-d16e-466b-9eff-02097008a8ec.png)

The _staging instance_ throws it as a "passed manual" :
![grafik](https://user-images.githubusercontent.com/8046960/175030066-4bc52696-86b2-4fab-883d-260b583731e2.png)

When I, for instance, check the two first codelist values:
`http://dd.eionet.europa.eu/vocabulary/biodiversity/n2000habitats/9110` and
`http://dd.eionet.europa.eu/vocabulary/biodiversity/n2000habitats/91E0`

There appears to be a redirect to the URLs ( where **vocabulary** seems to have changed now to **vocabularyconcept** and a **/view** added to the end of the url):
http://dd.eionet.europa.eu/vocabularyconcept/biodiversity/n2000habitats/9110/view
http://dd.eionet.europa.eu/vocabularyconcept/biodiversity/n2000habitats/91E0/view

However, when I look at the row **Concept URI** they the URIs are identical with the ones that are mapped in the dataset:
![grafik](https://user-images.githubusercontent.com/8046960/175033952-3b515c92-eb68-4ad1-af1a-68c659acefe0.png)

In my opinion the output of the staging instance should be correct, right? 
In addition, the error message in the production instance says to "Please check manually ......."
![grafik](https://user-images.githubusercontent.com/8046960/175035832-3e574a34-4aaa-460a-884f-531733935df9.png)

Furthermore, the  HabitatsDirectiveCodeValues (https://inspire.ec.europa.eu/codelist/HabitatsDirectiveCodeValue) are all still linked to the mapped URLs in the dataset.

Here the TestReports for both instances:
[test suite Annex III - Habitats and biotopes (HB).zip](https://github.com/INSPIRE-MIF/helpdesk-validator/files/8957936/test.suite.Annex.III.-.Habitats.and.biotopes.HB.zip)

Could you please advise how to proceed.
Kind Regards.

**Comment by fabiovinci** [comment, marked as answer]
Dear @Dnoack49,

it seems that the error was due to the lack of access to codelist.

![image](https://user-images.githubusercontent.com/22212406/175489408-203a6966-c9fc-4337-b782-e16ab6f7baf5.png)

It was a temporary unavailability since now the error is not present, see [the report](https://inspire.ec.europa.eu/validator/test-run/details.html?id=EID6d089e64-f8a8-4f15-b270-7a5d092e2f2f) and the report in production is the same of the one in staging.

**Comment by Dnoack49** [comment]
Dear @fabiovinci 
I have validated the dataset twice. The first time it, again, lead to an error (here [the report](https://inspire.ec.europa.eu/validator/test-run/details.html?id=EIDc121d5c6-a7f2-4af8-93d4-e2f432218cc2)).
The second time, however, the result was as you described (see [report](https://inspire.ec.europa.eu/validator/test-run/details.html?id=EID9560c054-c615-4af1-bf53-7cc08f6b07cc)). 
Thank you for clarifying, the issue can be closed.

kind regards
